xgcm/grid.py

```diff
@@ -1558,6 +1568,219 @@ def _grid_func(self, funcname, da, axis, **kwargs):
 
         return out
 
+    def pad(self, *arrays, boundary_width=None, boundary=None, fill_value=None):
+        """
+        Pads the boundary of given arrays along given Axes, according to information in Axes.boundary.
+
+        Parameters
+        ----------
+        arrays : Sequence[xarray.DataArray]
+            Arrays to pad according to boundary and boundary_width.
+        boundary_width : Dict[str: Tuple[int, int]
+            The widths of the boundaries at the edge of each array.
+            Supplied in a mapping of the form {axis_name: (lower_width, upper_width)}.
+        boundary : {None, 'fill', 'extend', 'extrapolate', dict}, optional
+            A flag indicating how to handle boundaries:
+            * None: Do not apply any boundary conditions. Raise an error if
+              boundary conditions are required for the operation.
+            * 'fill':  Set values outside the array boundary to fill_value
+              (i.e. a Dirichlet boundary condition.)
+            * 'extend': Set values outside the array to the nearest array
+              value. (i.e. a limited form of Neumann boundary condition.)
+            * 'extrapolate': Set values by extrapolating linearly from the two
```

Comment:
Did you figure out a way to implement `extrapolate`? Otherwise I think we should not mention it in the docstring?

Reply:
I haven't tried `extrapolate` yet. I have other problems with that docstring though - for example the way that `None` really means `"periodic"`...

Reply:
But that is `None` too?